### PR TITLE
Refactoring System Registration

### PR DESCRIPTION
**Pull request by StephenHodgson**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

Refactoring the core system registration for the XRTK
Also removed the core system accessors in favor of the `TryGetSystem<T>` which is now cached for better perf in hot loops

## Changes
- Fixes: #763
- Fixes: #764 

## Submodule changes

- [Examples](https://github.com/XRTK/Examples/pull/30)
- [Oculus](https://github.com/XRTK/Oculus/pull/106)
- [Lumin](https://github.com/XRTK/Lumin/pull/100)
- [SDK](https://github.com/XRTK/SDK/pull/227)
- [WMR](https://github.com/XRTK/WindowsMixedReality/pull/123)

### CODE REVIEW

File: XRTK-Core/Packages/com.xrtk.core/Runtime/Services/MixedRealityToolkit.cs [hunk 1]
```diff
@@ -407,111 +407,18 @@ private void InitializeServiceLocator()
...
                 {
-                    Debug.LogError("Failed to start the Camera System!");
-                }
-            }
-
-            if (ActiveProfile.IsInputSystemEnabled)
-            {
-                if (TryCreateAndRegisterService<IMixedRealityInputSystem>(ActiveProfile.InputSystemType, out var service, ActiveProfile.InputSystemProfile) && InputSystem != null)
-                {
-                    if (TryCreateAndRegisterService<IMixedRealityFocusProvider>(ActiveProfile.InputSystemProfile.FocusProviderType, out _))
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Runtime/Services/MixedRealityToolkit.cs:428]: This one might be tricky to figure out. We do have some special edge cases when we register some of our systems.

@SimonDarksideJ @FejZa

File: XRTK-Core/Packages/com.xrtk.core/Runtime/Services/MixedRealityToolkit.cs [hunk 2]
```diff
@@ -407,111 +407,18 @@ private void InitializeServiceLocator()
...
-                {
-                    Debug.LogError("Failed to start the Boundary System!");
-                }
-            }
-
-            if (ActiveProfile.IsSpatialAwarenessSystemEnabled)
-            {
-#if UNITY_EDITOR
-                // Setup the default spatial awareness layers in the project settings.
-                LayerExtensions.SetupLayer(31, Definitions.SpatialAwarenessSystem.MixedRealitySpatialAwarenessSystemProfile.SpatialAwarenessMeshesLayerName);
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Runtime/Services/MixedRealityToolkit.cs:460]: @FejZa we probably need to do something similar to how we setup the input axes but for the spatial awareness layers instead.

**Review by FejZa — APPROVED** [review]
Nice job on this! Looks good. I like the new service registration / retrieveal APIs.

File: XRTK-Core/Packages/com.xrtk.core/Runtime/Services/MixedRealityToolkit.cs [hunk 3]
```diff
@@ -11,13 +11,6 @@
 using XRTK.Definitions.Platforms;
 using XRTK.Extensions;
 using XRTK.Interfaces;
-using XRTK.Interfaces.BoundarySystem;
```
Comment (SimonDarksideJ) [inline XRTK-Core/Packages/com.xrtk.core/Runtime/Services/MixedRealityToolkit.cs:14]: I love this. #EndofLine

**Review by SimonDarksideJ — APPROVED** [review]

**Pull request merged**